ouroboros-consensus-shelley/src/Ouroboros/Consensus/Shelley/ShelleyHFC.hs

```diff
@@ -359,6 +360,8 @@ translateShelleyTables ::
      ( SL.TranslateEra     era TxOutWrapper
      , SL.TranslationError era TxOutWrapper ~ Void
      , EraCrypto (SL.PreviousEra era) ~ EraCrypto era
+     , Eq (Cardano.Ledger.Core.TxOut (SL.PreviousEra era))
```

Comment:
Same question. Why do we need these Eq constraints?

Reply:
> I think these can be removed as well

Again, I stand corrected. See https://github.com/input-output-hk/ouroboros-network/pull/4036#discussion_r991936963